CMakeLists.txt

```diff
@@ -156,7 +156,15 @@ include_directories(
 option(BUILD_PROOF_OF_SPACE_STATICALLY "Build ProofOfSpace target statically" OFF)
 IF (BUILD_PROOF_OF_SPACE_STATICALLY)
   message("Statically build ProofOfSpace")
-  target_link_libraries(ProofOfSpace -static -Wl,--whole-archive -lrt -lpthread -Wl,--no-whole-archive)
+  target_link_libraries(ProofOfSpace PUBLIC -static -Wl,--whole-archive -lrt -lpthread -Wl,--no-whole-archive)
+ENDIF()
+
+option(BUILD_STATIC_CHIAPOS_LIBRARY "Build chiapos library statically" OFF)
+IF (BUILD_STATIC_CHIAPOS_LIBRARY)
+  message("Statically build chiapos library")
+  add_library(chiapos_static STATIC src/chacha8.c src/verifier.hpp c-bindings/wrapper.cpp)
```

Comment:
you're adding a `.hpp` file as a source file here. That looks a bit suspicious. If it's correct, I think it warrants a comment. Does `wrapper.cpp` not include this header file?